sycl/include/CL/sycl/detail/pi.h

```diff
@@ -979,9 +979,9 @@ __SYCL_EXPORT pi_result piextQueueCreateWithNativeHandle(
 //
 // Memory
 //
-__SYCL_EXPORT pi_result piMemBufferCreate(pi_context context,
-                                          pi_mem_flags flags, size_t size,
-                                          void *host_ptr, pi_mem *ret_mem);
+__SYCL_EXPORT pi_result piMemBufferCreate(
+    pi_context context, pi_mem_flags flags, size_t size, void *host_ptr,
+    pi_mem *ret_mem, const cl_mem_properties_intel *properties = nullptr);
```

Comment:
```suggestion
    pi_mem *ret_mem, const pi_mem_properties *properties = nullptr);
```

Reply:
Please remove direct OpenCL usage from PI API.

Reply:
Just do double check, there is no such thing as `pi_mem_properties`, so you mean to define it in detail/pi.h, right? I think it's a good point, for now we can leave just as `using pi_mem_properties = pi_bitfield;` without any masks definitions.

Yet I'm a little bit confused right now. From https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/USM/cl_intel_unified_shared_memory.asciidoc `cl_mem_properties_intel` is used for USM extension and from the code, I guess, it's mapped on `pi_usm_mem_properties`. I don't want to reuse `pi_usm_mem_properties` as what I'm trying to achieve has nothing to do with USM, but does it make sense to have both?

Reply:
Okay, after more detailed look at the code, I see no direct mapping between `cl_mem_properties_intel` and `pi_usm_mem_properties`.

Reply:
> Just do double check, there is no such thing as pi_mem_properties, so you mean to define it in detail/pi.h, right?

Correct.

Reply:
Done